impress-org/givewp-next-gen
# Feature: Add recurring UI to donation form

Depends on #128, #137

## Description



This is the UI portion for recurring donations in our next gen donation form.  The form builder settings for recurring donations will be reflected on the donation form in an interactive and visual manner.

## Affects



This mainly affects the amount field rendering and how we are creating the field api nodes from the block.

## Visuals



<img width="796" alt="Screenshot 2023-03-10 at 11 41 33 AM" src="https://user-images.githubusercontent.com/10138447/224372851-29051236-c1b8-4d07-a5cb-6c0f47cbf213.png">


## Testing Instructions



- Pull branch and run `npm install && npm run build`
- Create a new Next Gen Form
- In the amount block settings - enable recurring
- Play around with the recurring settings and view the donation form to see the UI updates.
- Complete the donation form using the next gen test gateway for the full experience

## Pre-review Checklist
